services/workflows-service/src/transaction/transaction.service.ts

```diff
@@ -37,8 +40,16 @@ export class TransactionService {
           correlationId: transaction.transactionCorrelationId,
         });
       } catch (error) {
+        let errorToLog: Error = new Error('Unknown error', { cause: error });
+
+        if (error instanceof Prisma.PrismaClientKnownRequestError && error.code === 'P2002') {
+          errorToLog = new Error('Transaction already exists', { cause: error });
+        } else {
+          this.sentry.captureException(errorToLog);
```

Comment:
it would be better just to capture the original error instead yours

Reply:
As long as I add the original error to the chain (`cause`), why does it matter?